src/github.com/matrix-org/dendrite/clientapi/storage/output_room_events_table.go

```diff
@@ -72,7 +72,7 @@ func (s *outputRoomEventsStatements) Events(txn *sql.Tx, eventIDs []string) ([]g
 	}
 	defer rows.Close()
 
-	result := make([]gomatrixserverlib.Event, len(eventIDs))
+	var result []gomatrixserverlib.Event
```

Comment:
It does work, but when there is a mismatch then it panics rather than returning a sensible error message, which is also why I edited the error message.